Add `coverage_map` crate to provide `CoverageMapBuilder` and `CoverageMap` data structures

maplant opened the pull request:
## What does this new crate provide?

- `CoverageMapBuilder`: A data structure for keeping track of the current rank of all of the radios providing coverage to a given set of hexes. This data structure does not allow you to map radios to their ranked hex coverage, but allows you to build the `CoverageMap` data structure which does. Additionally, you can create a "submap" from a `CoverageMapBuilder` and a collection of `CoverageObject`s, which allows for inserting ranking only those hexes which are relevant to the set of `CoverageObject`s provided.
- `CoverageMap`: A data structure that maps hotspots to all of their ranked hex coverage.

## What work is left to do in this crate?

This crate is basically complete, but the following work remains:

- [x] Port over the old coverage tests from the mobile verifier.
- [x] Add new tests for the `submap` method.

michaeldjeffrey commented on coverage_map/src/indoor.rs:
```diff
@@ -0,0 +1,250 @@
+            indoor_cbrs_coverage("2", SignalLevel::Low),
+            indoor_cbrs_coverage("3", SignalLevel::High),
+            indoor_cbrs_coverage("4", SignalLevel::Low),
+            indoor_cbrs_coverage("5", SignalLevel::None),
```
Can a `SignalLevel` for an indoor radio be `None`?
When we talked earlier I was led to believe it can only be `High` or `Low`.

maplant replied:
Yes, it can be High Low or None

michaeldjeffrey commented:
When there's a single radio covering 2 hexes, both hexes should come back with rank `1`.
Currently, they are ranked sequentially.

```rs
#[test]
    fn single_radio() {
        let mut indoor_coverage = IndoorCellTree::default();

        insert_indoor_coverage_object(
            &mut indoor_coverage,
            indoor_cbrs_coverage_with_loc(
                "1",
                Cell::from_raw(0x8c2681a3064d9ff).unwrap(),
                date(2022, 2, 2),
            ),
        );
        insert_indoor_coverage_object(
            &mut indoor_coverage,
            indoor_cbrs_coverage_with_loc(
                "1",
                Cell::from_raw(0x8c2681a3064dbff).unwrap(),
                date(2022, 2, 2),
            ),
        );

        let coverage = into_indoor_coverage_map(indoor_coverage, &NoBoostedHexes, Utc::now())
            .collect::<Vec<_>>();
        for ranked in coverage {
            println!("{ranked:?}");
        }
    }
```

maplant commented:
> When there's a single radio covering 2 hexes, both hexes should come back with rank `1`. Currently, they are ranked sequentially.
> 
> ```rust
> #[test]
>     fn single_radio() {
>         let mut indoor_coverage = IndoorCellTree::default();
> 
>         insert_indoor_coverage_object(
>             &mut indoor_coverage,
>             indoor_cbrs_coverage_with_loc(
>                 "1",
>                 Cell::from_raw(0x8c2681a3064d9ff).unwrap(),
>                 date(2022, 2, 2),
>             ),
>         );
>         insert_indoor_coverage_object(
>             &mut indoor_coverage,
>             indoor_cbrs_coverage_with_loc(
>                 "1",
>                 Cell::from_raw(0x8c2681a3064dbff).unwrap(),
>                 date(2022, 2, 2),
>             ),
>         );
> 
>         let coverage = into_indoor_coverage_map(indoor_coverage, &NoBoostedHexes, Utc::now())
>             .collect::<Vec<_>>();
>         for ranked in coverage {
>             println!("{ranked:?}");
>         }
>     }
> ```

Fixed and added as test
